posthog/clickhouse/migrations/0045_add_insertion_timestamp_to_events.py

```diff
@@ -0,0 +1,30 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+from posthog.client import sync_execute
+from posthog.models.event.sql import (
+    EVENTS_TABLE_JSON_MV_SQL,
+    KAFKA_EVENTS_TABLE_JSON_SQL,
+)
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+ADD_COLUMNS_BASE_SQL = """
+ALTER TABLE {table}
+ON CLUSTER '{cluster}'
+ADD COLUMN IF NOT EXISTS inserted_at DateTime64(6, 'UTC') DEFAULT NOW()
+"""
+
+
+def add_columns_to_required_tables(_):
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="events", cluster=CLICKHOUSE_CLUSTER))
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="writable_events", cluster=CLICKHOUSE_CLUSTER))
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="sharded_events", cluster=CLICKHOUSE_CLUSTER))
```

Comment:
Looking at the schema diff, it seems only sharded_events are updated, I presume it should also include a change to `events` as well as a minimum?

Reply:
If you are re-running all migrations to get a diff, migration 001 will add the column as the `EVENTS_TABLE_SQL` was modified too.

Reply:
Ah, true, I forgot to modify those as we are passing the new field in `extra_fields`. I'll add it to those tables too. Thanks for pointing this out.

Reply:
Added them. I'd prefer if we just had the raw queries, not a huge fan of all the Python code mixing, as you can't really tell what's running. Anyways, thanks for spotting this 👍

Reply:
Hah yes 100% I would totally love to go dumb here and be explicit.

Reply:
In some other parts of the app it makes more sense because we are approaching the DB in a more interactive fashion as users interact with PostHog. But migrations are intended to be deterministic and part of the deployment process, so the added complexity doesn't really make sense in this case.